### PR TITLE
Fix OpenSSL dependency issues

### PR DESCRIPTION
**Pull request by uklotzde**
The dependency on _reqwest 0.8_ caused issues with system dependencies on a specific OpenSSL version. These issues disappeared after upgrading to _reqwest 0.9_, which relies on _native-tls_ by default.

Other notable changes:
* Upgraded _edition_ to "2018"
* Adjusted an expected test result that changed slightly
* Reduced redundant code
* Removed unused dependencies

### CODE REVIEW

**Comment by urschrei** [comment]
Some of these changes conflict / overlap with #15. @groteworld Are you happy for that to merge first, and then we can look at this?

**Comment by uklotzde** [comment]
I should've checked for pending PRs first :see_no_evil: #15 might already solve our issues.

**Comment by groteworld** [comment]
I'm sorry @uklotzde, can you make a new branch with changes you want to make. @urschrei is right, this merge diff is too hard to follow now on what is happening. I appreciate the work and I do think some of the changes I am seeing are good to be in the project
